sno/merge.py

```diff
@@ -124,30 +125,30 @@ def move_repo_to_merging_state(repo, merge_index, merge_context, merge_message):
     merge_index - the MergeIndex containing the conflicts found.
     merge_context - the MergeContext object for the merge.
     merge_message - the commit message for when the merge is completed.
-    ancestor, ours, theirs - CommitWithReference objects.
     """
-    if is_ongoing_merge(repo):
-        raise InvalidOperation("A merge is already ongoing")
-
-    # These are specific to sno repositories
+    assert RepoState.get_state(repo) != RepoState.MERGING
     merge_index.write_to_repo(repo)
     merge_context.write_to_repo(repo)
     write_repo_file(repo, MERGE_MSG, merge_message)
+    assert RepoState.get_state(repo) == RepoState.MERGING
 
 
 def abort_merging_state(ctx):
     """
     Put things back how they were before the merge began.
     Tries to be robust against failure, in case the user has messed up the repo's state.
     """
-    repo = ctx.obj.repo
+    repo = ctx.obj.get_repo(allowed_states=RepoState.ALL_STATES)
     is_ongoing_merge = repo_file_exists(repo, MERGE_HEAD)
     # If we are in a merge, we now need to delete all the MERGE_* files.
     # If we are not in a merge, we should clean them up anyway.
     remove_all_merge_repo_files(repo)
+    assert RepoState.get_state(repo) != RepoState.MERGING
 
     if not is_ongoing_merge:
-        raise InvalidOperation('Repository is not in "merging" state.')
+        raise InvalidOperation(
+            '`sno merge --abort` only works when sno repo is in "merging" state.'
```

Comment:
I think I prefer the old message, because if you type `sno merge --abort` you probably already know it doesn't work unless you're in merging state, but you think you _are_ in merging state.  So if it tells that your assumption is incorrect then that's a more helpful message

Reply:
Added current state to message.